internal/bminventory/inventory.go

```diff
@@ -464,7 +464,7 @@ func (b *bareMetalInventory) DeregisterCluster(ctx context.Context, params insta
 	var cluster common.Cluster
 	log.Infof("Deregister cluster id %s", params.ClusterID)
 
-	if err := b.db.First(&cluster, identity.AddUserFilter(ctx, "id = ?"), params.ClusterID).Error; err != nil {
```

Comment:
Shouldn't this function be removed if it's not being used anymore?

Reply:
it is still used once in ListCluster() where you have to filter the result before sending it to the user